### ISSUE TITLE
VirtualBox fails to build kernel modules with dkms

### ISSUE THREAD

**Issue by UriHerrera** [Nitrux/nitrux-bug-tracker#131]
**Describe the bug**

During the installation process of VirtualBox using the installer from Oracle, the installer fails to build the VirtualBox kernel modules with the following error from the log file `/var/log/vbox-setup.log`

```
error: implicit declaration of function ‘skb_gso_segment’; did you mean ‘skb_gso_reset’? [-Werror=implicit-function-declaration]
```

**Steps to reproduce the bug**

1. Install VirtualBox using the installer by Oracle
2. The installer completes the installation of VirtualBox but fails to build the kernel modules

**Expected behavior**

The VirtualBox installer successfully builds the kernel modules.

**Additional context**

As indicated in this [bug report at the Gentoo bug tracker](https://bugs.gentoo.org/912168).

> VirtualBox modules were broken by the following commit in Linux 6.5 that was backported to Linux 6.4.10:
>
> net: move gso declarations and functions to their files
>
> https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.4.y&id=bbe07adbaf39c2c5a95c3ca7eb52b2119d50af7d
> 
> VirtualBox 7.0.10's VBoxNetFlt-linux.c does have an #include <net/gso.h>, but it's wrapped in a check for kernel 6.5.0; this needs to be changed to 6.4.10.

---

The solution is to modify the file `VBoxNetFlt-linux.c` in `/opt/VirtualBox/src/vboxhost/vboxnetflt/linux/` 

```
sudo sed -i 's+#if RTLNX_VER_MIN(6,5,0)+#if RTLNX_VER_MIN(6,4,10)+g' /opt/VirtualBox/src/vboxhost/vboxnetflt/linux/VBoxNetFlt-linux.c
```

With that said, _**this is not an issue, bug, or problem caused by Nitrux**_; this is a problem with VirtualBox.

_**Disclaimer:** We do not develop VirtualBox. Please inform its developers at their respective bug trackers if you have issues._

**Comment by UriHerrera** [comment]
I'm adding the issue here for reference. However, I'm closing it because we're not Oracle, and this is not a problem caused by us.

**Comment by UriHerrera** [comment]
Oracle has fixed this bug in VirtualBox 7.0.12.
